File: cpp_solutions/121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int ans = 0;
        int mins = prices[0];
        
        for (int x = 0; x < prices.size();x++){
            mins = min(mins,prices[x]);
            ans = max(ans, prices[x] - mins);
        }
        
        return ans;
    }
};

File: python3_solution_set0/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        ans = 0
        current = prices[0]
        for x in range(len(prices)):
            current = min(current,prices[x])
            ans = max(ans, prices[x] - current)

        return ans
        
